### PR TITLE
Enable Renovate `automerge`

### LINKED ISSUE #365: Enable Renovate automerge of minor and patch dependencies
Example: https://github.com/api3dao/oev-dashboard/pull/39

### PR DESCRIPTION
**Pull request by aquarat**
As requested by @Siegrift this enables Remnovate `automerge` for patch and minor dependency releases.

Adding Siegrift because he requested the change, bdrhn9 because AFAIK he's the major user of this repository and hiletmis because he's familiar with Renovate.

Closes https://github.com/api3dao/airseeker/issues/365

### CODE REVIEW

**Comment by hiletmis** [comment]
`automerge` should be moved to rules like;

```
    {
      "matchDepTypes": ["devDependencies"],
      "matchUpdateTypes": ["patch", "minor"],
      "schedule": ["before 4am on Monday"],
      "groupName": "non-major-dev-dependencies",
      "automerge": true
    },
    {
      "matchDepTypes": ["dependencies"],
      "matchUpdateTypes": ["patch", "minor"],
      "schedule": ["before 4am on Monday"],
      "groupName": "non-major-dependencies",
      "automerge": true
    }
```

all PRs including major ones may automerged otherwise

File: renovate.json [hunk 1]
```diff
@@ -21,8 +21,9 @@
   ],
   "rangeStrategy": "bump",
   "lockFileMaintenance": {
-    "enabled": false
+    "enabled": true,
+    "automerge": true
   },
-  "reviewers": ["Siegrift"],
+  "reviewers": ["Siegrift", "aquarat", "bdrhn9"],
```
Comment (Siegrift) [inline renovate.json]: Maybe just @bdrhn9 as a reviewer? Btw. updating this warrants a change in technical documentation.

**Comment by aquarat** [comment]
Thanks @hiletmis, should `automerge` for the parent scope then be turned off? ( e.g. `$.lockFileMaintenance.automerge = false`)

**Comment by hiletmis** [comment]
> Thanks @hiletmis, should `automerge` for the parent scope then be turned off? ( $.lockFileMaintenance.automerge = false)

Yes should be removed. (setting false will override not recommended)

File: renovate.json [hunk 2]
```diff
@@ -10,19 +10,21 @@
...
       "schedule": ["before 4am on Monday"],
-      "groupName": "non-major-dependencies"
+      "groupName": "non-major-dependencies",
+      "automerge": true
     }
   ],
   "rangeStrategy": "bump",
   "lockFileMaintenance": {
-    "enabled": false
+    "enabled": true
```
Comment (Siegrift) [inline renovate.json]: Any reason to turn this on?

File: renovate.json (hunk 1 above)
Comment (Siegrift) [reply renovate.json]: Let's wait for Bedirhan to confirm he's OK with it.

File: renovate.json (hunk 2 above)
Comment (aquarat) [reply renovate.json]: (Reverted, see here: https://github.com/api3dao/oev-searcher/pull/327#discussion_r1709732933 )

File: renovate.json (hunk 1 above)
Comment (bdrhn9) [reply renovate.json]: I'm okay with it.

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Pull request merged**